### PR TITLE
[ISSUE-841][REFACTOR] Migrate shuffle client side conf to Celeborn Configuration System

### LINKED ISSUE #841: [REFACTOR] Migrate shuffle client side conf to  Celeborn Configuration System
### Describe the intention you'd like
A clear and concise description of what you want to do.

/cc @who-need-to-know

/assign @who-can-help-you

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related issues.
close #841

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX @pan3793

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -140,11 +140,13 @@ public ShuffleClientImpl(CelebornConf conf, UserIdentifier userIdentifier) {
         new TransportContext(dataTransportConf, new BaseMessageHandler(), true);
     dataClientFactory = context.createClientFactory();
 
-    int retryThreadNum = CelebornConf.pushDataRetryThreadNum(conf);
-    pushDataRetryPool = ThreadUtils.newDaemonCachedThreadPool("Retry-Sender", retryThreadNum, 60);
+    int pushDataRetryThreadNum = conf.pushDataRetryThreadNum();
+    pushDataRetryPool =
+        ThreadUtils.newDaemonCachedThreadPool("Retry-Sender", pushDataRetryThreadNum, 60);
```
Comment (pan3793) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:145]: we'd better add a `celeborn-` or `clb-` prefix to the name of the thread created by celeborn, and please naming in lowercase

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 2]
```diff
@@ -140,11 +140,13 @@ public ShuffleClientImpl(CelebornConf conf, UserIdentifier userIdentifier) {
...
 
-    int splitPoolSize = CelebornConf.clientSplitPoolSize(conf);
-    partitionSplitPool = ThreadUtils.newDaemonCachedThreadPool("Shuffle-Split", splitPoolSize, 60);
+    int pushSplitPartitionNumThreads = conf.pushSplitPartitionNumThreads();
+    partitionSplitPool =
+        ThreadUtils.newDaemonCachedThreadPool("Shuffle-Split", pushSplitPartitionNumThreads, 60);
```
Comment (pan3793) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:149]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -379,6 +379,18 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
   //                      Client                        //
   // //////////////////////////////////////////////////////
   def shuffleWriterMode: String = get(SHUFFLE_WRITER_MODE)
+  def shuffleForceFallback: Boolean = get(SHUFFLE_FORCE_FALLBACK)
+  def maxPartitionNumSupported: Long = get(MAX_PARTITION_NUM)
+  def shuffleMetaServicePort: Int = {
+    get(SHUFFLE_META_SERVICE_PORT) match {
+      case port if port == 0 =>
+        logWarning(
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:387]: > how about moving to the definition?

How to do that?

[the remark this answers is not in the document]
Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:387]: by leveraging `.checkValue`

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java (hunk 2 above)
Comment (AngersZhuuuu) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -1368,17 +1397,50 @@ object CelebornConf extends Logging {
...
+    buildConf("celeborn.push.limit.inFlight.sleepInterval")
+      .withAlternative("rss.limit.inflight.sleep.delta")
+      .categories("client")
+      .doc("Sleep interval when check netty in-flight requests to be done.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("50ms")
+
+  val PUSH_RETRY_NUM_THREADS: ConfigEntry[Int] =
+    buildConf("celeborn.push.retry.numThread")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1428]: numThread => threads

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -1368,17 +1397,50 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.pushdata.retry.thread.num")
+      .categories("client")
+      .doc("Sleep interval when check netty in-flight requests to be done.")
+      .version("0.2.0")
+      .intConf
+      .createWithDefault(Math.max(8, Runtime.getRuntime.availableProcessors()))
+
+  val PUSH_SPLIT_PARTITION_NUM_THREADS: ConfigEntry[Int] =
+    buildConf("celeborn.push.splitPartition.numThreads")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1437]: ditto

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java (hunk 1 above)
Comment (AngersZhuuuu) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java]: done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -1519,13 +1562,33 @@ object CelebornConf extends Logging {
     conf.getTimeAsMs("rss.replicate.fastfail.duration", "60s")
   }
 
-  def maxPartitionNumSupported(conf: CelebornConf): Long = {
-    conf.getInt("rss.max.partition.number", 500000)
-  }
+  val SHUFFLE_FORCE_FALLBACK: ConfigEntry[Boolean] =
+    buildConf("celeborn.shuffle.force.fallback")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1557]:
```suggestion
    buildConf("celeborn.shuffle.forceFallback.enabled")
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 4 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 7]
```diff
@@ -1519,13 +1562,33 @@ object CelebornConf extends Logging {
...
+      .version("0.2.0")
+      .doc("Whether force fallback shuffle to Spark's default.")
+      .booleanConf
+      .createWithDefault(false)
 
-  def forceFallback(conf: CelebornConf): Boolean = {
-    conf.getBoolean("rss.force.fallback", false)
-  }
+  val MAX_PARTITION_NUM: ConfigEntry[Long] =
+    buildConf("celeborn.shuffle.max.partition.number")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1566]:
```suggestion
    buildConf("celeborn.shuffle.forceFallback.numPartitionsThreshold")
```

**Review by pan3793 — APPROVED** [review]

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 8]
```diff
@@ -1519,13 +1562,33 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.max.partition.number")
+      .categories("client")
+      .version("0.2.0")
+      .doc(
+        "Celeborn will only accept shuffle of partition number lower than this configuration value.")
+      .longConf
+      .createWithDefault(500000)
+
+  val SHUFFLE_META_SERVICE_PORT: ConfigEntry[Int] =
+    buildConf("celeborn.shuffle.metaService.port")
```
[the remark this answers is not in the document]
Comment (waitinfuture) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1586]: no we dont

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 9]
```diff
@@ -1519,13 +1563,43 @@ object CelebornConf extends Logging {
...
+      .version("0.2.0")
+      .doc("Port used by the LifecycleManager on the Driver.")
+      .intConf
+      .checkValue(
+        (port: Int) => {
+          if (port == 0) {
+            logWarning(
+              "The user specifies the port used by the LifecycleManager on the Driver, and its" +
+                s" values is $port, which may cause port conflicts and startup failure.")
+          }
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1598]: I guess the warning condition should be `port != 0` based on the message?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 10]
```diff
@@ -1638,21 +1708,14 @@ object CelebornConf extends Logging {
...
-  }
-
-  def clientSplitPoolSize(conf: CelebornConf): Int = {
-    conf.getInt("rss.client.split.pool.size", 8)
-  }
+  val SHUFFLE_PARTITION_TYPE: ConfigEntry[String] =
+    buildConf("celeborn.shuffle.partition.type")
+      .withAlternative("rss.partition.type")
+      .categories("client")
+      .doc("")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1715]: doc is missed

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 11]
```diff
@@ -1638,21 +1708,14 @@ object CelebornConf extends Logging {
...
+      .version("0.2.0")
+      .stringConf
+      .createWithDefault("reduce")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1718]: please add value check

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 9 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1598]: Yea, updated

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 12]
```diff
@@ -1368,17 +1389,59 @@ object CelebornConf extends Logging {
...
+    buildConf("celeborn.push.limit.inFlight.sleepInterval")
+      .withAlternative("rss.limit.inflight.sleep.delta")
+      .categories("client")
+      .doc("Sleep interval when check netty in-flight requests to be done.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("50ms")
+
+  val SORT_PUSH_MEMORY_THRESHOLD: ConfigEntry[Long] =
+    buildConf("celeborn.sort.push.memory.threshold")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1420]: How about
```suggestion
    buildConf("celeborn.push.sort.memory.threshold")
```
?

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1420]: Sort Based puhser

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 11 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1718]: Done

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**